src/traits/data.jl

```diff
@@ -136,17 +136,113 @@ Tables.materializer(D::Type{<:Data}) = D
 # DATAFRAME INTERFACE
 # --------------------
 
-function Base.getproperty(data::Data, col::Symbol)
-  if col == :geometry
+function Base.getproperty(data::Data, var::Symbol)
+  if var == :geometry
     domain(data)
   else
     cols = Tables.columns(values(data))
-    Tables.getcolumn(cols, col)
+    Tables.getcolumn(cols, var)
   end
 end
 
-Base.getproperty(data::Data, col::AbstractString) =
-  getproperty(data, Symbol(col))
+Base.getproperty(data::Data, var::AbstractString) =
+  getproperty(data, Symbol(var))
+
+function Base.getindex(data::Data, ind::Int, vars::AbstractVector{Symbol})
+  _checkvars(vars)
+  _rmgeometry!(vars)
+  dom   = domain(data)
+  table = values(data)
+  row   = Tables.subset(table, ind)
+  pairs = (var => Tables.getcolumn(row, var) for var in vars)
+  (; pairs..., geometry=dom[ind])
+end
+
+Base.getindex(data::Data, ind::Int, vars::AbstractVector{<:AbstractString}) =
+  getindex(data, ind, Symbol.(vars))
+
+function Base.getindex(data::Data, ind::Int, ::Colon)
+  dom   = domain(data)
+  table = values(data)
+  row   = Tables.subset(table, ind)
+  vars  = Tables.columnnames(row)
+  pairs = (var => Tables.getcolumn(row, var) for var in vars)
+  (; pairs..., geometry=dom[ind])
+end
+
+Base.getindex(data::Data, ind::Int, var::Symbol) =
+  getproperty(data, var)[ind]
+
+Base.getindex(data::Data, ind::Int, var::AbstractString) =
+  getindex(data, ind, Symbol(var))
+
+function Base.getindex(data::Data, inds::AbstractVector{Int}, vars::AbstractVector{Symbol})
+  _checkvars(vars)
+  _rmgeometry!(vars)
+  table = values(data)
+
+  newdom   = view(domain(data), inds)
+  subset   = Tables.subset(table, inds)
+  cols     = Tables.columns(subset)
+  pairs    = (var => Tables.getcolumn(cols, var) for var in vars)
+  newtable = (; pairs...) |> Tables.materializer(table)
+
+  vals = Dict(paramdim(newdom) => newtable)
+  constructor(data)(newdom, vals)
```

Comment:
This one can be written in terms of view

Reply:
In this case, the number of lines would be the same:
```julia
  dataview = view(data, inds)
  newdom   = domain(dataview)
  cols     = Tables.columns(values(dataview))
  pairs    = (var => Tables.getcolumn(cols, var) for var in vars)
  newtable = (; pairs...) |> Tables.materializer(table)